### ISSUE TITLE
Ability to manage errors in a custom way

### ISSUE THREAD

**Issue by dfernandezm** [phiz71/vertx-swagger#47]
I was trying work out a way of getting customizable errors from the SwaggerRouter when something fails. Right now it seems that it is only possible to get 400 Bad Request from any Runtime Exception, and any error on the handling of the request is automatically managed as well, from `com.github.phiz71.vertx.swagger.router.SwaggerRouter`:

```
 private static void configureRoute(Route route, String serviceId, Operation operation, EventBus eventBus, Function<RoutingContext, DeliveryOptions> configureMessage) {
...
        route.handler(context -> {
            try {
                JsonObject message = new JsonObject();
                operation.getParameters().forEach(parameter -> {
                    String name = parameter.getName();
                    Object value = PARAMETER_EXTRACTORS.get(parameter.getIn()).extract(name, parameter, context);
                    message.put(name, value);
                });

                // callback to configure message e.g. provide message header values
                DeliveryOptions deliveryOptions = configureMessage != null ? configureMessage.apply(context) : new DeliveryOptions();

                eventBus.<String> send(serviceId, message, deliveryOptions, operationResponse -> {
                    if (operationResponse.succeeded()) {
                        if (operationResponse.result().body() != null) {
                            vertxLogger.debug(operationResponse.result().body());
                            manageHeaders(context.response(), operationResponse.result().headers());
                            context.response().end(operationResponse.result().body());
                        } else {
                            context.response().end();
                        }
                    } else {
                        vertxLogger.debug("Internal Server Error", operationResponse.cause());
                        manageError((ReplyException)operationResponse.cause(), context.response());
                    }
                });
            } catch (RuntimeException e) {
                vertxLogger.debug("sending Bad Request", e);
                badRequestEnd(context.response());
            }

        });
```

I'd like to extend the `catch` bits and the `else` part  (when the operation does not succeed). What do you advise? This is a static method, the operations aren't abstract, I am guessing the only way is customizing it in a fork. Thanks.

**Comment by phiz71** [comment]
Actually, the `SwaggerRouter` behavior face to errors is to change the code and the status of the response.
That is done in the `manageError`.

It reads the error cause of the `message` and get a failure code and a failure message.

From Verticles point of view, message failures are caused by catching `xxxAPIExpcetion`. These custom exceptions can be generated by the codegen plugin and are based on the response part of each resources.
You can have a look at : 
https://github.com/phiz71/vertx-swagger/blob/master/sample/petstore-vertx-server/src/main/java/io/swagger/server/api/verticle/PetApiException.java
and
https://github.com/phiz71/vertx-swagger/blob/master/sample/petstore-vertx-server/src/main/java/io/swagger/server/api/verticle/PetApiVerticle.java#L182


All unexpected errors will send a 500-UnhandledException and will be logged.

**Comment by phiz71** [comment]
If you don't want to use the codegen plugin, just keep in mind that if you want the `SwaggerRouter` to answer a "412 -My Custom Message", then the message on the eventBus should be failed with this code and this message.

**Comment by dfernandezm** [comment]
Oh, I see, you've built that support already... I did not know about the example either, good to know!

In my rush to get most of it done this past week I think I redid that support, extending to an RxJava workflow in our business logic:

* error bubbles out from Observables chain to a single onError channel, whose handler fails the message in the eventBus
* Not knowing of your builtin support, I forked SwaggerRouter, and changed the catch block of RuntimeException to manage the possible badRequest. Did the same if the message was failed in the eventBus.

I understand you already have most of the support I required, will try to adapt back. Thanks for prompt answer.

**Comment by phiz71** [comment]
Hello. If it is OK for you, could you close the issue ?
Thanks.

**Comment by nitish24p** [comment]
Is there a way i can send this error as a custom json response, and not as status message

**Comment by phiz71** [comment]
> Is there a way i can send this error as a custom json response, and not as status message

Hi @nitish24p , as this issue is closed and you opened #109 , I will answer in your issue
